internal/errgroup/group_test.go

```diff
@@ -307,86 +328,78 @@ func Test_group_Limitation(t *testing.T) {
 		limit int
 	}
 	type fields struct {
-		egctx            context.Context
-		cancel           func()
-		wg               sync.WaitGroup
 		limitation       chan struct{}
 		enableLimitation atomic.Value
-		cancelOnce       sync.Once
-		mu               sync.RWMutex
-		emap             map[string]struct{}
-		errs             []error
-		err              error
 	}
 	type want struct {
+		want Group
 	}
 	type test struct {
 		name       string
 		args       args
 		fields     fields
 		want       want
-		checkFunc  func(want) error
+		checkFunc  func(want, Group) error
 		beforeFunc func(args)
 		afterFunc  func(args)
 	}
-	defaultCheckFunc := func(w want) error {
+	defaultCheckFunc := func(w want, g Group) error {
+		got, want := g.(*group), w.want.(*group)
+		if !reflect.DeepEqual(got.enableLimitation, want.enableLimitation) {
+			return errors.Errorf("got = %v, want %v", got, w.want)
+		}
+		if got.limitation != nil && want.limitation != nil {
+			return errors.Errorf("got = %v, want %v", got, w.want)
+		}
 		return nil
 	}
 	tests := []test{
-		// TODO test cases
-		/*
-		   {
-		       name: "test_case_1",
-		       args: args {
-		           limit: 0,
-		       },
-		       fields: fields {
-		           egctx: nil,
-		           cancel: nil,
-		           wg: sync.WaitGroup{},
-		           limitation: nil,
-		           enableLimitation: nil,
-		           cancelOnce: sync.Once{},
-		           mu: sync.RWMutex{},
-		           emap: nil,
-		           errs: nil,
-		           err: nil,
-		       },
-		       want: want{},
-		       checkFunc: defaultCheckFunc,
-		   },
-		*/
-
-		// TODO test cases
-		/*
-		   func() test {
-		       return test {
-		           name: "test_case_2",
-		           args: args {
-		           limit: 0,
-		           },
-		           fields: fields {
-		           egctx: nil,
-		           cancel: nil,
-		           wg: sync.WaitGroup{},
-		           limitation: nil,
-		           enableLimitation: nil,
-		           cancelOnce: sync.Once{},
-		           mu: sync.RWMutex{},
-		           emap: nil,
-		           errs: nil,
-		           err: nil,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
+		func() test {
```

Comment:
how about rewrite it like this? I think it is easier to read, what do you think?
```golang
357         {
358             name: "set disable when limit is 0",
359             args: args{
360                 limit: 0,
361             },
362             fields: fields{
363                 limitation:       make(chan struct{}),
364                 enableLimitation: atomic.Value{},
365             },
366             want: want{
367                 want: &group{
368                     enableLimitation: func() atomic.Value {
369                         var el atomic.Value
370                         el.Store(false)
371                         return el
372                     }(),
373                 },
374             },
375         },
```